pliers/extractors/audio.py

```diff
@@ -488,4 +494,88 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
     For details on argument specification visit:
     https://librosa.github.io/librosa/effect.html.'''
 
+
     _feature = 'percussive'
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+
+    ''' Extract probability of audio event classes based on AudioSet-YouTube 
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the window (in seconds) on which label extraction
+        is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If not defined, returns probabilities for all labels.
+    spectrogram (bool): if True, plots mel-spectrogram of the audio stimulus.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('params', 'top_n')
+
+    def __init__(self, hop_size=0.1, top_n=None, 
+                 spectrogram=False, **yamnet_kwargs):
+        verify_dependencies(['yamnet'])
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        WEIGHTS_PATH = path.join(MODULE_PATH, 'yamnet.h5')
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+
+        self.spectrogram = spectrogram
+        self.params = yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+
+        if yamnet_kwargs:
+            for par in self.params.__dict__:
+                if yamnet_kwargs[par]:
+                    setattr(self.params, par, yamnet_kwargs[par])
+
+        self.tf_graph = tf.Graph()
+        with self.tf_graph.as_default():
+            self.model = yamnet.yamnet_frames_model(self.params)
+            self.model.load_weights(WEIGHTS_PATH)
+
+        self.labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
+        self.params = {par: val for par,val in self.params.__dict__.items()
+                       if par.isupper()}
+        self.top_n = top_n if top_n else len(self.labels)
+        super(AudiosetLabelExtractor, self).__init__()
+
+    def _extract(self, stim):
+        
+        data = stim.data
+        self.params['SAMPLE_RATE'] = stim.sampling_rate
+        
+        with self.tf_graph.as_default():
+            preds, spectrogram = self.model.predict(np.reshape(data, [1,-1]), 
+                                                    steps=1)
+        
+        if self.spectrogram:
```

Comment:
Yeah, don't think this belong here. Instead, the user is free to inspect the `Extractor` and inspect/plt the spectrogram.

This could maybe be a misc tool, but its easy enough to do I'm not too worried.

Reply:
removed plotting option and flag, now spectrogram is retrievable from Extractor object.